tests/test_client.py

```diff
@@ -49,3 +52,37 @@ def test_invalid_credentials(
 
         with pytest.raises(error_name):
             client.delete_documents(consent_id=consent_id)
+
+
+@pytest.mark.parametrize('content', [
+    __file__,
+    Path(__file__),
+    open(__file__, 'rb'),
+    open(__file__, 'r'),
+    open(__file__),
+    Path(__file__).open(),
+    Path(__file__).open('rb'),
+    Path(__file__).open('r'),
+    Path(__file__).read_bytes(),
+    Path(__file__).read_text().encode(),
+    b64encode(Path(__file__).read_bytes()),
+    bytearray(Path(__file__).read_bytes()),
+])
+def test_parse_content(content):
+    expected_result = b64encode(Path(__file__).read_bytes()).decode()
+    result = parse_content(content)
+    assert result == expected_result
+
+
+@pytest.mark.parametrize(('content', 'error'), [
+    (uuid.uuid4().hex, FileNotFoundError),
+    (Path(uuid.uuid4().hex), FileNotFoundError),
+    (Path(__file__).read_bytes().decode(), OSError),
```

Comment:
OSError seems weird here?

Reply:
Any other suggestions?

Reply:
What about FileNotFoundError?

Reply:
We will keep it This way as the OSError contains enough useful information